amgut/db/patches/0041.sql

```diff
@@ -208,60 +262,231 @@ CREATE INDEX idx_dna_plate_extraction_robot_id ON pm.dna_plate ( extraction_robo
 CREATE INDEX idx_dna_plate_extraction_kit_lot_id ON pm.dna_plate ( extraction_kit_lot_id );
 CREATE INDEX idx_dna_plate_extraction_tool_id ON pm.dna_plate ( extraction_tool_id );
 
-CREATE TABLE pm.protocol_targeted (
-    protocol_targeted_id      bigserial  NOT NULL,
-    name                      varchar  NOT NULL,
-    updated_on                timestamp DEFAULT current_timestamp NOT NULL,
-    barcode_sequence_plate_id bigint  ,
-    master_mix_lot_id         bigint  ,
-    water_lot_id              bigint  ,
-    tm300_8_tool_id           bigint  ,
-    tm50_8_tool_id            bigint  ,
-    processing_robot_id       bigint  ,
-    CONSTRAINT pk_protocol_targeted PRIMARY KEY ( protocol_targeted_id ),
-    CONSTRAINT fk_protocol_targeted FOREIGN KEY ( barcode_sequence_plate_id ) REFERENCES pm.barcode_sequence_plate( barcode_sequence_plate_id )    ,
-    CONSTRAINT fk_protocol_targeted_mm FOREIGN KEY ( master_mix_lot_id ) REFERENCES pm.master_mix_lot( master_mix_lot_id )    ,
-    CONSTRAINT fk_protocol_targeted_wl FOREIGN KEY ( water_lot_id ) REFERENCES pm.water_lot( water_lot_id )    ,
-    CONSTRAINT fk_protocol_targeted_tm300 FOREIGN KEY ( tm300_8_tool_id ) REFERENCES pm.tm300_8_tool( tm300_8_tool_id )    ,
-    CONSTRAINT fk_protocol_targeted_tm50 FOREIGN KEY ( tm50_8_tool_id ) REFERENCES pm.tm50_8_tool( tm50_8_tool_id )    ,
-    CONSTRAINT fk_protocol_targeted_robot FOREIGN KEY ( processing_robot_id ) REFERENCES pm.processing_robot( processing_robot_id )
- );
-CREATE INDEX idx_protocol_targeted_barcode_sequence_plate_id ON pm.protocol_targeted ( barcode_sequence_plate_id );
-CREATE INDEX idx_protocol_targeted_master_mix_lot_id ON pm.protocol_targeted ( master_mix_lot_id );
-CREATE INDEX idx_protocol_targeted_water_lot_id ON pm.protocol_targeted ( water_lot_id );
-CREATE INDEX idx_protocol_targeted_tm300_8_tool_id ON pm.protocol_targeted ( tm300_8_tool_id );
-CREATE INDEX idx_protocol_targeted_tm50_8_tool_id ON pm.protocol_targeted ( tm50_8_tool_id );
-CREATE INDEX idx_protocol_targeted_processing_robot_id ON pm.protocol_targeted ( processing_robot_id );
-
-CREATE TABLE pm.library_plate (
-    library_plate_id     bigserial  NOT NULL,
+CREATE TABLE pm.target_gene_plate (
+    target_gene_plate_id bigserial  NOT NULL,
     name                 varchar  NOT NULL,
-    email                varchar  ,
-    created_on           timestamp  ,
+    email                varchar  NOT NULL,
+    created_on           timestamp  NOT NULL,
     dna_plate_id         bigint  NOT NULL,
-    protocol_id          bigint  NOT NULL,
-    protocol_targeted_id bigint  ,
+    barcode_sequence_plate_id bigint  NOT NULL,
```

Comment:
I wonder if it's necessary to make those properties `NOT NULL`. What if the user has to enter a plate without be aware of a specific parameter? (e.g., trying to document some old records)